apps/recommendation/api/src/huggy/core/endpoint/retrieval_endpoint.py

```diff
@@ -213,6 +232,35 @@ async def model_score(self) -> t.List[RecommendableItem]:
             for r in prediction_result.predictions
         ]
 
+    async def model_score(self) -> t.List[RecommendableItem]:
+        instance = self.get_instance(self.size)
+
+        if self.cached:
+            instance_hash = self.__get_instance_hash(instance)
+            cache_key = f"{self.endpoint_name}:{instance_hash}"
+            result = await VERTEX_CACHE.get(cache_key)
+            if result is not None:
+                self._log_cache_usage(cache_key, "Used")
+                return result
+
+        result = await self._vertex_retrieval_score(instance)
+
+        if self.cached:
+            await VERTEX_CACHE.set(cache_key, result)
+            self._log_cache_usage(cache_key, "Set")
```

Comment:
peut être mettre des petits commentaires pour détailler le flux 

#Retrieve cache if exists
...

#Compute retrieval if cache not found or used
...

#Update Cache
...